commandapi-platforms/commandapi-bukkit/commandapi-bukkit-vh/src/main/java/dev/jorel/commandapi/CommandAPIVersionHandler.java

```diff
@@ -49,11 +49,13 @@ public interface CommandAPIVersionHandler {
 	 * @return an instance of NMS which can run on the specified Minecraft version
 	 */
 	static CommandAPIPlatform<?, ?, ?> getPlatform() {
+		CommandAPIPlatform<?, ?, ?> latestNMS = new NMS_1_21_R1();
```

Comment:
Constructing `latestNMS` here means it always attempts to load when this method is called. This seems to cause an exception on any version other than 1.21/1.21.1 (tested on Paper 1.19.4 and 1.20).

For example, simply adding the CommandAPI plugin to the server and running on Paper 1.20 gives this error:

```
[17:07:23 ERROR]: [CommandAPI] Error initializing plugin 'CommandAPI-9.6.0-SNAPSHOT_12_Aug_2024_(08-42-37PM).jar' in folder 'plugins' (Is it up to date?)
java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    dev/jorel/commandapi/nms/NMS_1_21_R1.<clinit>()V @38: invokestatic
  Reason:
    Type 'net/minecraft/core/IRegistryCustom$Dimension' (current frame, stack[0]) is not assignable to 'net/minecraft/core/HolderLookup$a'
  Current Frame:
    bci: @38
    flags: { }
    locals: { 'org/bukkit/craftbukkit/v1_21_R1/CraftServer', 'org/bukkit/Server' }
    stack: { 'net/minecraft/core/IRegistryCustom$Dimension', 'net/minecraft/world/flag/FeatureFlagSet' }
  Bytecode:
    0000000: b804 c04c 2bc1 04c2 9900 272b c004 c24b
    0000010: 2ab6 0703 b608 752a b607 03b6 0876 b906
    0000020: 8401 00b6 07a1 b808 7bb3 007f a700 0701
    0000030: b300 7f12 e813 087d 1308 7d12 f0b8 0881
    0000040: b300 e113 0344 1308 8313 0885 b808 8bb3
    0000050: 034a 1308 8d13 088e 1308 8fb8 088b b306
    0000060: 6c12 4313 0890 b608 9457 043b a700 064c
    0000070: 033b 1ab3 0711 b1                      
  Exception Handler Table:
    bci [97, 108] => handler: 111
    bci [97, 108] => handler: 111
  Stackmap Table:
    same_frame(@47)
    same_frame(@51)
    same_locals_1_stack_item_frame(@111,Object[#386])
    append_frame(@114,Integer)

        at dev.jorel.commandapi.CommandAPIVersionHandler.getPlatform(CommandAPIVersionHandler.java:52) ~[CommandAPI-9.6.0-SNAPSHOT_12_Aug_2024_(08-42-37PM).jar:?]
        at dev.jorel.commandapi.CommandAPI.onLoad(CommandAPI.java:112) ~[CommandAPI-9.6.0-SNAPSHOT_12_Aug_2024_(08-42-37PM).jar:?]
        at dev.jorel.commandapi.CommandAPIMain.onLoad(CommandAPIMain.java:68) ~[CommandAPI-9.6.0-SNAPSHOT_12_Aug_2024_(08-42-37PM).jar:?]
        at io.papermc.paper.plugin.storage.ServerPluginProviderStorage.processProvided(ServerPluginProviderStorage.java:59) ~[paper-1.20.jar:git-Paper-17]
        at io.papermc.paper.plugin.storage.ServerPluginProviderStorage.processProvided(ServerPluginProviderStorage.java:18) ~[paper-1.20.jar:git-Paper-17]
        at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:40) ~[paper-1.20.jar:git-Paper-17]
        at io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler.enter(LaunchEntryPointHandler.java:36) ~[paper-1.20.jar:git-Paper-17]
        at org.bukkit.craftbukkit.v1_20_R1.CraftServer.loadPlugins(CraftServer.java:430) ~[paper-1.20.jar:git-Paper-17]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:273) ~[paper-1.20.jar:git-Paper-17]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1101) ~[paper-1.20.jar:git-Paper-17]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318) ~[paper-1.20.jar:git-Paper-17]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

This prevents the CommandAPI from loading. Perhaps instead of constructing the nms instance to check `NMS#compatibleVersions`, this code could just check if the version matches a hardcoded current latest minor version. This would need to be updated manually, but that seems similar to needing to update the result given by `shouldUseLatestNMSVersion`.

```java
if (shouldUseLatestNMSVersion) {
    return new NMS_1_21_R1();
}

switch (version) {
	case "1.16.5" -> new NMS_1_16_R3();
	// ...
	case "1.21", "1.21.1" -> new NMS_1_21_R1();
    default -> {
        if (shouldBeMoreLenientForMinorVersions && version.split("\\.")[1].equals("21")) {
            yield new NMS_1_21_R1();
        } else {
            throw UnsupportedVersionException(version);
        }
    }
}
```

Just need to avoid constructing `NMS_1_21_R1`  when the switch statement will match a `version` String.